### PR TITLE
component "tip of the day" added to homepage with random tip  

### LINKED ISSUE #249: Accessibility "tip of the day"
This feature is an enhancement/nice to have idea for consideration

**What:**  When a user hits the site, a random tip or "did you know" style bit of information is displayed with a link to the relevant article/info on the website

**Why:** Could be good for a user interested in accessible web dev but not sure where to start or for someone who already knows some things, but is perhaps fed a new idea/tool/tip they weren't aware of and maybe wouldn't search for.

**UI Idea:** Up for discussion, but could be at the top in a banner or somewhere else prominent so it catches the eye.

**Implementation:** Could pull from existing articles on the site if the content makes sense or create a dataset of prompts and their relevant articles which is more manual but you would have better control over the content and experience. ex. "Did you know XX percent of adults in the US have difficulty with low contrast? CTA to an article on designing beautiful high contrast UI"

### PR DESCRIPTION
**Pull request by Neha**
## Describe your changes
- Used already avilable tipoftheday component to display on homepage
- Created  hard coded 7 tips in `data/tipsOfTheDay.js`
- Updated the component to render random tip from the data
- Added CSS to make it 100% on mobile screen

## Screenshots 
Screenshot 2023-05-10 at 7 13 32 AM

Screenshot 2023-05-10 at 7 18 06 AM

## Link to issue

Closes #249 

## Checklist before requesting a review

- [X] I have performed a self-review of my code.
- [X] Followed the repository's [Contributing Guidelines](/CONTRIBUTING.md).
- [X] I ran the app and tested it locally to verify that it works as expected.
- [X] I have checked my code with an automatic accessibility tool such as Axe Dev Tools or Wave 
      and it shows no errors.

## Additional Information (Optional)
Any additional information that you want to give us.

### CODE REVIEW

File: components/TipOfTheDay/index.tsx [hunk 1]
```diff
@@ -1,11 +1,34 @@
+import { useEffect, useState } from "react"
 import styles from "./TipOfTheDay.module.css"
+import { TIPS } from "../../data/tipsOfTheDay"
```
Comment (EmmaDawsonDev) [inline components/TipOfTheDay/index.tsx:3]: Please rename this to use lowercase

File: components/TipOfTheDay/index.tsx [hunk 2]
```diff
@@ -1,11 +1,34 @@
...
-			<a href="#">Name of Link</a>
-		</div>
-	)
+interface ITip {
+	title: string
+	description: string
+	tipLink: string
+}
+
+export const TipOfTheDay: React.FC = () => {
```
Comment (EmmaDawsonDev) [inline components/TipOfTheDay/index.tsx:11]:
```suggestion
export const TipOfTheDay = () => {
```
No need to specify the type, it's inferred by typescript

File: components/TipOfTheDay/index.tsx [hunk 3]
```diff
@@ -1,11 +1,34 @@
...
+	const [randomIndex, setRandomIndex] = useState<number>(0)
+
+	useEffect(() => {
+		const TIPS_LENGTH = TIPS.length
```
Comment (EmmaDawsonDev) [inline components/TipOfTheDay/index.tsx:15]: I don't think there's a need to set this as a variable, just use the array length directly.

File: components/TipOfTheDay/index.tsx [hunk 4]
```diff
@@ -1,11 +1,34 @@
...
+		const randomNumber = Math.floor(Math.random() * TIPS_LENGTH)
+
+		setRandomIndex(randomNumber)
+	})
+
+	const createTipOfTheDay = () => {
+		const tipIndex: ITip = TIPS[randomIndex]
```
Comment (EmmaDawsonDev) [inline components/TipOfTheDay/index.tsx:22]: This should be moved outside of the function.

File: components/TipOfTheDay/index.tsx [hunk 5]
```diff
@@ -1,11 +1,34 @@
...
+
+export const TipOfTheDay: React.FC = () => {
+	const [randomIndex, setRandomIndex] = useState<number>(0)
+
+	useEffect(() => {
+		const TIPS_LENGTH = TIPS.length
+		const randomNumber = Math.floor(Math.random() * TIPS_LENGTH)
+
+		setRandomIndex(randomNumber)
+	})
```
Comment (EmmaDawsonDev) [inline components/TipOfTheDay/index.tsx:19]: This useEffect is missing a dependency array.

File: components/TipOfTheDay/index.tsx [hunk 6]
```diff
@@ -1,11 +1,34 @@
...
+			<>
+				<h2>{tipIndex.title}</h2>
+				<p>{tipIndex.description}</p>
+				<a target="_blank" rel="noreferrer" href={tipIndex.tipLink}>
+					Read More
+				</a>
+			</>
+		)
+	}
+	return <div className={styles.card}>{createTipOfTheDay()}</div>
```
Comment (EmmaDawsonDev) [inline components/TipOfTheDay/index.tsx:33]: Calling functions inside of the component leads to flashing when the component renders on screen. If you try refreshing the page you'll see that the tip flashes a couple of times as things are causing it to render more than once.

It's better to call the function and store it in a variable, then pass only the variable into the component.

File: data/tipsOfTheDay.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
+export interface ITips {
+	title: string
+	description: string
+	tipLink: string
+}
+
+export const TIPS: ITips[] = [
```
Comment (EmmaDawsonDev) [inline data/tipsOfTheDay.ts:7]: Please use lowercase

File: components/TipOfTheDay/index.tsx [hunk 8]
```diff
@@ -1,11 +1,34 @@
...
+
+	const createTipOfTheDay = () => {
+		const tipIndex: ITip = TIPS[randomIndex]
+		return (
+			<>
+				<h2>{tipIndex.title}</h2>
+				<p>{tipIndex.description}</p>
+				<a target="_blank" rel="noreferrer" href={tipIndex.tipLink}>
+					Read More
+				</a>
```
Comment (EmmaDawsonDev) [inline components/TipOfTheDay/index.tsx:29]: If this opens in a new window we need to specify that clearly for the user. Also, `read more` as a link text is very generic. Either it needs an aria label which adds the extra "Read more about ...tip topic" or we do something like make the heading itself into a link.

**Review by EmmaDawsonDev — CHANGES_REQUESTED** [review]
Love these tips Neha! Thank you for taking the time to add them!

There's a few changes to make to how the card is rendered, then this should be ready to merge. Let me know if anything is unclear of if you need any help with anything.

**Review by EmmaDawsonDev — CHANGES_REQUESTED** [review]
Love these tips Neha! Thank you for taking the time to add them!

There's a few changes to make to how the card is rendered, then this should be ready to merge. Let me know if anything is unclear of if you need any help with anything.

File: data/tipsOfTheDay.ts (hunk 7 above)
Comment (Neha) [reply data/tipsOfTheDay.ts:7]: As I was using dummy content now I kept it in uppercase but I will change to lowercase now

**Comment by Neha** [comment]
> Love these tips Neha! Thank you for taking the time to add them!
> 
> There's a few changes to make to how the card is rendered, then this should be ready to merge. Let me know if anything is unclear of if you need any help with anything.

Thank you @EmmaDawsonDev . All the changes are done. Let me know if any other change is required.

**Review by EmmaDawsonDev — APPROVED** [review]
This looks great Neha, thank you!

**Pull request merged**